changelog/assert_ctfe.dd

```diff
@@ -0,0 +1,20 @@
+ctfe-only function elision
+
+As of now assert(__ctfe) inside the outmoster scope of a function body or inside an in contract will cause the function to not be emitted into the object file.
+Should a another function attempt to use a ctfe-only function a compile error will be emitted.
+
+-------
+string generateMixin(string name)
+{
+    assert(__ctfe);
+    return `string `" ~ name ~ `= "This is not a good mixin";`;
```

Comment:
I think you probably don't want the `"` that comes right before `~ name ~` ... ?

Reply:
Given that this is an example it might be best to make it as simple as possible, something like:

```D
return "My name is " ~ name;
```

;-)